core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresQueryDao.scala

```diff
@@ -149,26 +151,41 @@ private[r2dbc] class PostgresQueryDao(settings: R2dbcSettings, connectionFactory
     .bind(2, toSequenceNr)
     .bind(3, settings.querySettings.bufferSize)
 
-  protected val allPersistenceIdsSql =
-    sql"SELECT DISTINCT(persistence_id) from $journalTable ORDER BY persistence_id LIMIT ?"
-
-  protected val persistenceIdsForEntityTypeSql =
-    sql"SELECT DISTINCT(persistence_id) from $journalTable WHERE persistence_id LIKE ? ORDER BY persistence_id LIMIT ?"
+  protected def allPersistenceIdsSql = {
+    // FIXME
+    require(
+      settings.numberOfDataPartitions == 1,
+      "allPersistenceIdsSql not implemented for more than one data-partition yet")
```

Comment:
these queries that are not for a specific slice would have to be implemented by several queries to each data partition and then merge the results